src/basic/io-util.h

```diff
@@ -77,8 +77,12 @@ static inline bool FILE_SIZE_VALID_OR_INFINITY(uint64_t l) {
 #define IOVEC_NULL (struct iovec) {}
 #define IOVEC_INIT(base, len) { .iov_base = (base), .iov_len = (len) }
 #define IOVEC_MAKE(base, len) (struct iovec) IOVEC_INIT(base, len)
-#define IOVEC_INIT_STRING(string) IOVEC_INIT((char*) string, strlen(string))
-#define IOVEC_MAKE_STRING(string) (struct iovec) IOVEC_INIT_STRING(string)
+#define IOVEC_INIT_STRING(string)               \
+        ({                                      \
+                char *_s = (char*) (string);    \
+                IOVEC_MAKE(_s, strlen(_s));     \
+        })
+#define IOVEC_MAKE_STRING(string) IOVEC_INIT_STRING(string)
```

Comment:
then maybe drop IOVEC_INIT_STRING() altogether?

btw, the reason these used to be two macros: 

in C99 this is not OK:

```c
struct foo bar = (struct foo) { … };
```

this however is:

```c
struct foo bar;
bar = (struct foo) { … };
```

because one is an initialization and the other an assignment.

GNU removed the distinction, and allows both. this code comes from a time we weren#t relying on that gnu extension yet. If we do now, then mabye just drop the IOVEC__INIT_STRING() macro altoegther and just keep IOVEC_MAKE_STRING()?